### PR TITLE
fix #8128: addon-centered clashing with addon-docs for react

### LINKED ISSUE #8128: Addon-centered breaks layout in addon-docs
**Describe the bug**
When using addon-centered together with addon-docs it creates two problems:

- ~addon-docs suddenly says that it no longer finds a component~
- It still renderes the component centered in the page as an overlay over the docs page.

**To Reproduce**
Setup addon-centered together with addon-docs.

**Expected behavior**
I expect addons-centered not to affect addon-docs.

**Screenshots**
Screenshot 2019-09-19 at 12 43 06
Screenshot 2019-09-19 at 12 43 12

**System:**
```
Environment Info:

  System:
    OS: macOS 10.14.6
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
  Binaries:
    Node: 12.10.0 - /usr/local/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 6.11.3 - /usr/local/bin/npm
  Browsers:
    Chrome: 76.0.3809.132
    Safari: 12.1.2
  npmPackages:
    @storybook/addon-a11y: ^5.2.0 => 5.2.0
    @storybook/addon-actions: ^5.2.0 => 5.2.0
    @storybook/addon-centered: ^5.2.0 => 5.2.0
    @storybook/addon-docs: ^5.2.1 => 5.2.1
    @storybook/addon-knobs: ^5.2.0 => 5.2.0
    @storybook/addon-links: ^5.2.0 => 5.2.0
    @storybook/addons: ^5.2.0 => 5.2.0
    @storybook/react: ^5.2.0 => 5.2.0
```

### PR DESCRIPTION
**Pull request by mrijke**
The centered addon was clashing when used in combination with the
docs addon. When on the docs page, the centered style should not be applied.
Note that this commit only fixes it for React.

Issue:
See #8128, when on the DocsPage of the docs addon, the component is being centered in the page.

## What I did
Adapted the suggestion from https://github.com/storybookjs/storybook/issues/8128#issuecomment-538975286 to check if the current view is the Docs view.

Note that this only fixes the React variant of this addon. I don't have much experience with the other frameworks, but if requested I wouldn't mind trying out fixing those as well.
Also had to disable the eslint check for no-undef for the `window` usage, tried to make it a bit better with checking if `window` is not undefined, but open for better suggestions :)

### CODE REVIEW

**Comment by shilman** [comment]
Hmmm @CodeByAlex were you looking at this at one point? This seems like a pretty hacky solution.

**Comment by CodeByAlex** [comment]
I was looking into ways to optimize addon centered but not this specific issue. Whatever solution we come up with, the same should be applied to this issue for zooming #8389

**Comment by mrijke** [comment]
@shilman Yeah I agree, this is not the most beautiful solution, but I was hitting this issue and saw people proposed something, just wanted to start a PR to get the ball rolling.
I'm all open for other suggestions, if I have some more time I'll try and see if I can somehow better determine whether the docs view is active.
Or if you have another idea of how to approach this problem, please share some pointers :)

File: addons/centered/src/react.tsx [hunk 1]
```diff
@@ -5,6 +5,17 @@ import parameters from './parameters';
 import styles from './styles';
 
 function centered(storyFn: () => ReactNode) {
+  /* eslint-disable no-undef */
```
Comment (ndelangen) [inline addons/centered/src/react.tsx]: Perhaps a `useViewmode` hook would be a good idea?

**Comment by loonywizard** [comment]
Hi! Looks like this PR is ready, do you have any plans for releasing it? In our [project](https://github.com/kupibilet-frontend/ui) we are waiting for this change

This is not urgent for us, but we're looking forward to use `addon-centered` in our Storybook

Thank you!

**Comment by ndelangen** [comment]
@shilman considering the centered decorator will be likely deprecated in 6.0.0 for this:
https://github.com/storybookjs/storybook/pull/9229

We might as well fix this in 5.3 maybe?

**Comment by shilman** [comment]
@ndelangen I'm fine with patching this into 5.3 after the release is out

File: addons/centered/src/react.tsx (hunk 1 above)
Comment (shilman) [reply addons/centered/src/react.tsx]: @ndelangen I think that would be very useful, independent of this PR

**Review by ndelangen — APPROVED** [review]

**Pull request merged**